### PR TITLE
Homework # 3 is ready for pull request

### PR DESCRIPTION
**Pull request by lusinehovhan**

### CODE REVIEW

File: src/com/company/hashmap_enum/company/enumMonths/Months.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    public int getNumberOfHolidays() {
+        return numberOfHolidays;
+    }
+
+    public int getNumberOfDays() {
+        return numberOfDays;
+    }
+
+    public void printHolidayMarch(String holidayMarch) {
```
Comment (aram-hakobyan) [inline src/com/company/hashmap_enum/company/enumMonths/Months.java]: Let's make this method return boolean value, which will tell whether passed dayOfMonth is a holiday. Do not forget to rename the method.

File: src/com/company/hashmap_enum/company/hashMap/Main.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.company.hashmap_enum.company.hashMap;
+
+public class Main {
+
+    public static void main(String[] args) {
+        StudentService studentService = new StudentService();
+        studentService.fillStudentList();
```
Comment (aram-hakobyan) [inline src/com/company/hashmap_enum/company/hashMap/Main.java]: What if we call frequencyStudents() before fillStudentList()? Let's create students list inside StudentService() constructor.

File: src/com/company/hashmap_enum/company/hashMap/StudentService.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        frequencyMap.put(0, student1);
+        frequencyMap.put(1, student2);
+        frequencyMap.put(2, student3);
+        frequencyMap.put(3, student4);
+        frequencyMap.put(4, student5);
+        frequencyMap.put(5, student6);
+
+    }
+
+    public void frequencyStudents() {
```
Comment (aram-hakobyan) [inline src/com/company/hashmap_enum/company/hashMap/StudentService.java]: As per task this method must return a map of students, so let's rewrite it and rename as well.

File: src/com/company/hashmap_enum/company/hashMap/StudentService.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                map.put(s, 1);
+            }
+        }
+        for (String s : map.keySet()) {
+            int occurrence = map.get(s);
+            System.out.println(s + " occur " + occurrence + " time(s).");
+        }
+    }
+
+    public void numberOSstudentsInEachFaculty() {
```
Comment (aram-hakobyan) [inline src/com/company/hashmap_enum/company/hashMap/StudentService.java]: Same for this method.

File: src/com/company/hashmap_enum/company/enumMonths/Months.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
+package com.company.hashmap_enum.company.enumMonths;
+
+import java.util.Arrays;
+
+enum Months {
+    JANUARY(new String[]{"January 1", "January 6", "January 28"}, 3, 31),
```
Comment (aram-hakobyan) [inline src/com/company/hashmap_enum/company/enumMonths/Months.java]: It's better to use an integer array for holidays that holds the number of days, e.g. {4,9,15}, without month name and this string stuff.

File: src/com/company/hashmap_enum/company/enumMonths/Months.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+    public int getNumberOfHolidays() {
+        return numberOfHolidays;
+    }
+
+    public int getNumberOfDays() {
+        return numberOfDays;
+    }
+
+    public boolean dayOfMonth(String holidayMarch) {
```
Comment (aram-hakobyan) [inline src/com/company/hashmap_enum/company/enumMonths/Months.java]: Please rename method (e.g. isHoliday) and deal with integers, not strings as holidays.

File: src/com/company/hashmap_enum/company/hashMap/StudentService.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+                occurrence++;
+                map.put(s, occurrence);
+            } else {
+                map.put(s, 1);
+            }
+        }
+        return map;
+    }
+
+    public Map getFacultyMap(List<Student> studentsList) {
```
Comment (aram-hakobyan) [inline src/com/company/hashmap_enum/company/hashMap/StudentService.java]: fillStudentList() should not be here.

File: src/com/company/hashmap_enum/company/hashMap/StudentService.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        frequencyMap.put(0, student1);
+        frequencyMap.put(1, student2);
+        frequencyMap.put(2, student3);
+        frequencyMap.put(3, student4);
+        frequencyMap.put(4, student5);
+        frequencyMap.put(5, student6);
+    }
+
+
+    public Map getStudentsMap(List<Student> studentsList) {
```
Comment (aram-hakobyan) [inline src/com/company/hashmap_enum/company/hashMap/StudentService.java]: Call fillStudentList() inside StudentService  constructor.

File: src/com/company/hashmap_enum/company/hashMap/Student.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
+package com.company.hashmap_enum.company.hashMap;
+
+public class Student {
+    private String firstName;
+    private String lastName;
+    private int age;
+    private String phoneNumber;
+    private String faculty;
+
+    public Student(String firstName, String llastName, int age, String phoneNumber, String faculty) {
```
Comment (apatian) [inline src/com/company/hashmap_enum/company/hashMap/Student.java]: fix the typo - should be lastName, not llastName

File: src/com/company/hashmap_enum/company/hashMap/Main.java [hunk 10]
```diff
@@ -0,0 +1,10 @@
+package com.company.hashmap_enum.company.hashMap;
+
+public class Main {
+
+    public static void main(String[] args) {
+        StudentService studentService = new StudentService();
```
Comment (aram-hakobyan) [inline src/com/company/hashmap_enum/company/hashMap/Main.java]: If students list is already inside StudentsService, there is no need to pass it as a parameter.

File: src/com/company/hashmap_enum/company/hashMap/StudentService.java [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+        Student student6 = new Student("Elen", "Mirzoyan", 12, "093333333", "English");
+
+        students.add(student1);
+        students.add(student2);
+        students.add(student3);
+        students.add(student4);
+        students.add(student5);
+        students.add(student6);
+    }
+
```
Comment (aram-hakobyan) [inline src/com/company/hashmap_enum/company/hashMap/StudentService.java]: Constructor must be above all methods.

File: src/com/company/hashmap_enum/company/hashMap/StudentService.java [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+    public StudentService() {
+        fillStudentList();
+    }
+
+    public Map getStudentsMap(List students) {
+        Map<String, Integer> map = new HashMap<String, Integer>();
+        String[] studentsNames = new String[students.size()];
+        for (int i = 0; i < students.size(); i++) {
+            map.put(students.get(i).toString(), 0);
```
Comment (aram-hakobyan) [inline src/com/company/hashmap_enum/company/hashMap/StudentService.java]: Try to find a solution without using this array.

**Pull request closed without merging**